services/electiontrigger/timer_based_election_trigger.go

```diff
@@ -56,7 +58,12 @@ func (t *TimerBasedElectionTrigger) RegisterOnElection(blockHeight primitives.Bl
 
 	t.triggerCancelled = make(chan struct{})
 	t.timer = time.AfterFunc(timeout, func() {
-		t.onTimerTimeout()
+		h, v, eh, tc := t.loadTriggerContextUnderLock()
```

Comment:
I think you can simplify this by accessing local variables from the closure directly

Reply:
we can simplify this but:
1. we need to read lock access to `t`
1. if we read lock in this scope it will include the attempt to write into the elections channel which may cause a lock to be kept more than we must.

this is why I think it's a little better this way. but i'm not against moving the argument `t` in `loadTruggerContextUnderLock` from the receiver position to a parameter to indicate it's a helper function, what do you think?